lib/Auth/AuthHandler.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Spark\Auth;
+
+use Auryn\Injector;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class AuthHandler
+{
+
+    protected $authenticator;
+    protected $injector;
+
+    public function __construct(AbstractAuthenticator $authenticator)
+    {
+        $this->authenticator = $authenticator;
+    }
+
+    public function __invoke(Request $request, Response $response, callable $next)
+    {
+
+        $token = $this->authenticator->getToken($request);
+
+        if ($token && !$this->authenticator->isValid($token)) {
+            throw new \Exception($this->authenticator->getErrorMessage());
```

Comment:
Personally I lean towards Authenticator being stateless and passing the token as a method parameter. If it was done stateful, you would way to pass some kind of DTO through from the Request to Authenticator. Depends on if we foresee a need to authenticate more than one token in the lifetime of a single request.